src/main.rs

```diff
@@ -346,7 +347,23 @@ enum Subcommand {
     CoinswapSend,
 }
 
+static INIT: Once = Once::new();
+
+/// Setup function that will only run once, even if called multiple times.
+fn setup_logger() {
+    INIT.call_once(|| {
+        env_logger::builder()
+            .is_test(true)
+            .filter_module("teleport", log::LevelFilter::Trace)
+            .init();
+    });
+}
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    setup_logger();
+
+    log::trace!("Hello");
```

Comment:
nit: remove or replace with a more descriptive log startup message

Reply:
Ya this shouldn't be there. Removed.